contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -181,13 +182,56 @@ const char *bbf_servicename = "MSSQLSERVER";
 char	   *bbf_language = "us_english";
 #define MD5_HASH_LEN 32
 
-
 Datum
 trancount(PG_FUNCTION_ARGS)
 {
 	PG_RETURN_UINT32(NestedTranCount);
 }
 
+Datum
+babelfish_concat_wrapper(PG_FUNCTION_ARGS)
+{
+	text		*arg1, *arg2, *new_text;
+	int32		arg1_size, arg2_size, new_text_size;
+	bool		first_param = PG_ARGISNULL(0);
+	bool		second_param = PG_ARGISNULL(1);
+	
+	if (pltsql_concat_null_yields_null)
+	{
+		if(first_param || second_param)
+		{
+			PG_RETURN_NULL(); // If any is NULL, return NULL
+		}
+	}
+	else
+	{
+		if (first_param && second_param)
+		{
+			PG_RETURN_NULL(); // If both are NULL, return NULL
+		}
+		else if (second_param)
+		{
+			PG_RETURN_TEXT_P(PG_GETARG_TEXT_PP(0)); // If only the second string is NULL, return the first string
+		}
+		else if (first_param)
+		{
+			PG_RETURN_TEXT_P(PG_GETARG_TEXT_PP(1)); // If only the first string is NULL, return the second string
+		}
+	}
+	arg1 = PG_GETARG_TEXT_PP(0);
+	arg2 = PG_GETARG_TEXT_PP(1);
+	arg1_size = VARSIZE_ANY_EXHDR(arg1);
+	arg2_size = VARSIZE_ANY_EXHDR(arg2);
+	new_text_size = arg1_size + arg2_size + VARHDRSZ;
+	new_text = (text *) palloc(new_text_size);
+
+	SET_VARSIZE(new_text, new_text_size);
+	memcpy(VARDATA(new_text), VARDATA_ANY(arg1), arg1_size);
+	memcpy(VARDATA(new_text) + arg1_size, VARDATA_ANY(arg2), arg2_size);
```

Comment:
seems like this implementation is inspired by `text_catenate` but lets put assert here to check arg1_size and arg2_size > 0? what would happen if empty string is passed? and is there existing tests cases to validate this change?